src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -13,8 +13,14 @@ namespace AutoRest.Swagger.Model.Utilities
 {
     public static class ValidationUtilities
     {
-        private static readonly string XmsPageable = "x-ms-pageable";
+        private const string XmsPageable = "x-ms-pageable";
         private static readonly Regex TrackedResRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private const string AzureHostSignature = "management.azure.com";
+
+        // if x-ms-parameterized-host extension is used or if the host is anything of the form management.azure, assume this to be an ARMServiceDefinition
+        // TODO: refine this definition for azure data plain also check x-ms-parameterized-host in detail to ensure this is ARM
+        public static bool IsARMServiceDefinition(ServiceDefinition serviceDefinition) => 
```

Comment:
note that virtually all validation rules are meant for ARM, so this smells like a solo effort for something that should be handled in general? note that autorest no longer runs validation rules by default anyways